services/src/main/java/org/keycloak/services/clientpolicy/executor/ConfidentialClientAcceptExecutor.java

```diff
@@ -45,6 +45,7 @@ public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyExcep
         switch (context.getEvent()) {
             case AUTHORIZATION_REQUEST:
             case TOKEN_REQUEST:
+            case SERVICE_ACCOUNT_TOKEN_REQUEST:
```

Comment:
As you've recognized, the beginning of `clientCredentialsGrant()` checks whether the client that sent a request is confidential client or not. However, this check is a small task and it does not affect performance.

Reply:
Yes, I've recognized this, but I've added the check to ConfidentialClientAcceptExecutor just in case that we add some more checks in ConfidentialClientAcceptExecutor in the future, which would not be in the `clientCredentialsGrant()` . That is why I prefered to rather add this.

Reply:
@mposolda Yes, I agree with you.